python/lib/physiological.py

```diff
@@ -580,6 +586,150 @@ def insert_event_file(self, event_data, event_file, physiological_file_id,
+        into the physiological_task_event table, linking it to the
+        physiological file ID already inserted in physiological_file.
+
+        TODO: NEW method. See the previous iteration named
+        `insert_event_file_legacy`. This should be the one and only
+        method once we are sure every data will not be impacted in db.
+        Only called in `eeg.py`.
+        :param event_data           : list with dictionaries of events
+                                      information to insert into
+                                      physiological_task_event
+         :type event_data           : list
+        :param event_file           : name of the event file
+         :type event_file           : str
+        :param physiological_file_id: PhysiologicalFileID to link the event info to
+         :type physiological_file_id: int
+        :param blake2               : blake2b hash of the task event file
+         :type blake2               : str
+        """
+
+        # event_file_id = self.physiological_event_file_obj.insert(
+        #     physiological_file_id,
+        #     'tsv',
+        #     event_file
+        # )
+
+        event_fields = (
+            'PhysiologicalFileID', 'Onset',     'Duration',   'TrialType',
+            'ResponseTime',        'EventCode', 'EventValue', 'EventSample',
+            'EventType',           'FilePath'
+        )
+        # known opt fields
+        optional_fields = (
+            'trial_type',  'response_time', 'event_code',
+            'event_value', 'event_sample',  'event_type',
+            'value',       'sample',        'duration',
+            'onset'
+        )
+        # all listed fields
+        known_fields = {*event_fields, *optional_fields}
+
+        for row in event_data:
+            # nullify not present optional cols
+            for field in optional_fields:
+                if field not in row.keys():
+                    row[field] = None
+
+            # has additional fields?
+            additional_fields = {}
+            for field in row:
+                if field not in known_fields and row[field].lower() != 'nan':
+                    additional_fields[field] = row[field]
+
+            # get values of present optional cols
+            onset = 0
+            if isinstance(row['onset'], (int, float)):
+                onset = row['onset']
+            else:
+                # try casting to float, cannot be n/a
+                # should raise an error if not a number
+                onset = float(row['onset'])
+
+            duration = 0
+            if isinstance(row['duration'], (int, float)):
+                duration = row['duration']
+            else:
+                try:
+                    # try casting to float
+                    duration = float(row['duration'])
+                except ValueError:
+                    # value could be 'n/a'
+                    # put it at 0
```

Comment:
it does not look like duration is set to 0 here instead of n/a. Unless I am misunderstanding the comment?

Reply:
My comment is not clear. The variable `duration` is set to 0 before the if/else.